cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityStatusStore.java

```diff
@@ -0,0 +1,236 @@
+ */
+public class CapabilityStatusStore implements CapabilityReader, CapabilityWriter {
+
+  private static final Gson GSON = new Gson();
+  private final TransactionRunner transactionRunner;
+
+  @Inject
+  CapabilityStatusStore(TransactionRunner transactionRunner) {
+    this.transactionRunner = transactionRunner;
+  }
+
+  /**
+   * Return the current status for a capability. If capability is not present, throws {@link IllegalArgumentException}
+   *
+   * @param capability
+   * @return {@link CapabilityStatus}
+   */
+  public CapabilityStatus getStatus(String capability) throws IOException {
+    return TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable capabilityTable = context.getTable(StoreDefinition.CapabilitiesStore.CAPABILITIES);
+      Collection<Field<?>> keyField = Collections
+        .singleton(Fields.stringField(StoreDefinition.CapabilitiesStore.NAME_FIELD, capability));
+      Collection<String> returnField = Collections.singleton(StoreDefinition.CapabilitiesStore.STATUS_FIELD);
+      Optional<StructuredRow> result = capabilityTable.read(keyField, returnField);
+      return result.map(structuredRow -> CapabilityStatus
+        .valueOf(structuredRow.getString(StoreDefinition.CapabilitiesStore.STATUS_FIELD).toUpperCase())).orElse(null);
+    }, IOException.class);
+  }
+
+  @Override
+  public boolean isEnabled(String capability) throws IOException {
+    return getStatus(capability) == CapabilityStatus.ENABLED;
+  }
+
+  @Override
+  public void checkAllEnabled(Collection<String> capabilities) throws IOException, CapabilityNotAvailableException {
+    TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable capabilityTable = context.getTable(StoreDefinition.CapabilitiesStore.CAPABILITIES);
+      List<List<Field<?>>> multiKeys = new ArrayList<>();
+      capabilities.forEach(capability -> {
+        multiKeys
+          .add(Collections.singletonList(Fields.stringField(StoreDefinition.CapabilitiesStore.NAME_FIELD, capability)));
+      });
+      Map<String, String> capabilityMap = capabilityTable.multiRead(multiKeys).stream().
+        collect(Collectors.toMap(
+          structuredRow -> structuredRow.getString(StoreDefinition.CapabilitiesStore.NAME_FIELD),
+          structuredRow -> structuredRow.getString(StoreDefinition.CapabilitiesStore.STATUS_FIELD)
+        ));
+      for (String capability : capabilities) {
+        if (!capabilityMap.containsKey(capability) || CapabilityStatus
+          .valueOf(capabilityMap.get(capability).toUpperCase()) != CapabilityStatus.ENABLED) {
+          throw new CapabilityNotAvailableException(capability);
+        }
+      }
+    }, IOException.class, CapabilityNotAvailableException.class);
+  }
+
+  @Override
+  public void checkAllEnabled(ApplicationSpecification appSpec) throws IOException, CapabilityNotAvailableException {
+    Set<String> capabilities = appSpec.getPlugins().entrySet().stream()
```

Comment:
You can move this to the `CapabilityReader` interface as the default implementation.

Reply:
Done